web/modules/custom/dpl_patron_menu/dpl_patron_menu.module

```diff
@@ -31,3 +33,57 @@ function dpl_patron_menu_preprocess_page(array &$variables): void {
     'menu' => $plugin_block->build(),
   ];
 }
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ *
+ * @param array<mixed> $form
+ *   An associative array containing the structure of the form.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The current state of the form.
+ */
+function dpl_patron_menu_form_dpl_library_agency_list_size_settings_form_alter(array &$form, FormStateInterface $form_state): void {
+  $config = \Drupal::service('config.factory')
+    ->getEditable('dpl_library_agency.list_size_settings');
+
+  $form['menu_list_size_settings'] = [
+    '#type' => 'fieldset',
+    '#title' => t('Menu list size settings', [], ['context' => 'List size (settings)']),
+    '#description' => t('The number of items to display in the menu list.', [], ['context' => 'List size (settings)']),
+    '#tree' => FALSE,
+  ];
+
+  $form['menu_list_size_settings']['menu_list_size_desktop'] = [
+    '#type' => 'number',
+    '#title' => t('Menu list size on desktop', [], ['context' => 'List size (settings)']),
+    '#default_value' => $config->get('menu_list_size_desktop') ?? DplMenuSettings::MENU_LIST_SIZE_DESKTOP,
+    '#min' => 1,
+    '#step' => 1,
+  ];
+
+  $form['menu_list_size_settings']['menu_list_size_mobile'] = [
+    '#type' => 'number',
+    '#title' => t('Menu list size on mobile', [], ['context' => 'List size (settings)']),
+    '#default_value' => $config->get('menu_list_size_mobile') ?? DplMenuSettings::MENU_LIST_SIZE_MOBILE,
+    '#min' => 1,
+    '#step' => 1,
+  ];
+
+  $form['#submit'][] = 'dpl_patron_menu_handler_submit';
+}
+
+/**
+ * Custom submit callback.
+ *
+ * @param array<mixed> $form
+ *   An associative array containing the structure of the form.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The current state of the form.
+ */
+function dpl_patron_menu_handler_submit(array &$form, FormStateInterface $form_state): void {
+  \Drupal::service('config.factory')
```

Comment:
You should use config service here instead.

Reply:
I tried implementing the use of config service here, but only with limited success.
Config service will not let me use `->set()` function, as the config is immutable.

Do you have another suggestion?